sys/kern/link_elf.c

```diff
@@ -1877,7 +1919,12 @@ elf_lookup(linker_file_t lf, Elf_Size symidx, int deps, Elf_Addr *res)
 			*res = 0;
 			return (EINVAL);
 		}
-		*res = ((Elf_Addr)ef->address + sym->st_value);
+#ifdef __CHERI_PURE_CAPABILITY__
+		*res = (uintcap_t)make_capability(sym, ef->address +
+		    sym->st_value);
+#else
+		*res = ((vaddr_t)ef->address + sym->st_value);
```

Comment:
Should probably switch to `ptraddr_t`

Reply:
Oops, yes.  I went ahead and fixed it now.  There are some other stragglers in this file we will have to fix when merging.